test/e2e/network_segmentation.go

```diff
@@ -508,6 +509,84 @@ var _ = Describe("Network Segmentation", func() {
 		expectedMessage := fmt.Sprintf("primary network already exist in namespace %q: %q", f.Namespace.Name, primaryNadName)
 		Expect(actualConditions[0].Message).To(Equal(expectedMessage))
 	})
+
+	Context("pod2Egress on a user defined primary network", func() {
+		const (
+			externalContainerName = "ovn-k-egress-test-helper"
+		)
+		var externalIpv4 string
+		BeforeEach(func() {
+			externalIpv4, _ = createClusterExternalContainer(
+				externalContainerName,
+				"registry.k8s.io/e2e-test-images/agnhost:2.45",
+				runExternalContainerCmd(),
+				httpServerContainerCmd(port),
+			)
+
+			DeferCleanup(func() {
+				deleteClusterExternalContainer(externalContainerName)
+			})
+		})
+
+		DescribeTable(
+			"can be accessed to from the pods running in the Kubernetes cluster",
+			func(netConfigParams networkAttachmentConfigParams, clientPodConfig podConfiguration) {
+				netConfig := newNetworkAttachmentConfig(netConfigParams)
+
+				netConfig.namespace = f.Namespace.Name
+				clientPodConfig.namespace = f.Namespace.Name
+
+				By("creating the attachment configuration")
+				_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Create(
+					context.Background(),
+					generateNAD(netConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("instantiating the client pod")
+				clientPod, err := cs.CoreV1().Pods(clientPodConfig.namespace).Create(
+					context.Background(),
+					generatePodSpec(clientPodConfig),
+					metav1.CreateOptions{},
+				)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(clientPod).NotTo(BeNil())
+
```

Comment:
check that the Pod is connected to the UDN before you check the network connectivity

Reply:
lemme see how the other tests are doing this..

Reply:
what did you decide here? We could check the annotations on the pod and verify its default route is via the UDN network?

Reply:
@tssurya Do you mind taking care of this one please?

Reply:
ack I can I don't see other tests doing this but this should be easy.. lemme see if I can push a fix for this

Reply:
This is done!